### PR TITLE
More tracking!

### PR DESCRIPTION
**Pull request by addshore**
https://phabricator.wikimedia.org/T117414

This adds tracking to:
 - Download button
 - Copy button
 - Start of the process (entering a URL) as well as the URL itself, as well as if it is a wikipage or direct image
 - Errors encountered when trying to enter a URL
 - Completion of every step of the process (using the step name)
 - Completion of the process / landing on the final page

### CODE REVIEW

File: redesign/js/app/views/DialogueEvaluationView.js [hunk 1]
```diff
@@ -37,6 +39,10 @@ $.extend( DialogueEvaluationView.prototype, {
 	},
 
 	_copyAttribution: function( trigger ) {
+		var self = this;
+
+		self._tracking.trackEvent( 'Button', 'CopyAttributation' );
```
Comment (wmde-manicki) [inline redesign/js/app/views/DialogueEvaluationView.js]: `CopyAttribution`?

**Comment by wmde-manicki** [comment]
Looks good although I haven't tested it yet as my local Piwik doesn't want to cooperate

**Comment by addshore** [comment]
Okay, so the download link tracking doesnt work.
I'm guessing thats down to $( '.track-click' ).click( function() { being in main.js and not called after we build the download button in the dom.

Solution 1: Call this thing after every time we change the Dom?
Solution 2: Do the tracking for this link in another way.

The copy tracking also doesn't quite work:

11:24 AM <jakob_WMDE> for the clipboard thing we can just add a self variable to the render method
11:25 AM <jakob_WMDE> and then instead of the current { text: this._copyAttribution } ); we do { text: function(trigger) { self._copyAttribution(trigger); self._track... } } );

**Comment by jakobw** [comment]
In `main.js` changing the first line of the click tracking thing from
`$( '.track-click' ).click( function() {`
to
`$( document ).on( 'click', '.track-click', function() {`
fixes that issue.

**Comment by jakobw** [comment]
Fixed the download button and copy button tracking!

**Comment by wmde-manicki** [comment]
Now those two work two. Great!

**Pull request merged**